playground/backend/internal/executors/executor.go

```diff
@@ -76,8 +76,12 @@ func (ex *Executor) Validate() func(chan bool, chan error, *sync.Map) {
 }
 
 // Prepare returns the function that applies all preparations of executor
-func (ex *Executor) Prepare() func(chan bool, chan error) {
-	return func(doneCh chan bool, errCh chan error) {
+func (ex *Executor) Prepare() func(chan bool, chan error, *sync.Map, string) {
+	return func(doneCh chan bool, errCh chan error, valRes *sync.Map, filePath string) {
+		isKata, ok := valRes.Load(validators.KatasValidatorName)
+		if ok && isKata.(bool) {
+			ex.preparators = *preparators.GetJavaPreparatorsForKatas(filePath)
+		}
```

Comment:
It doesn't seem a good idea to add here Java-specific logic. What do you think about the creation of the `processValidationResults` function in `code_processing.go` that will change the preparators according to the SDK and validation results?

Reply:
Done.